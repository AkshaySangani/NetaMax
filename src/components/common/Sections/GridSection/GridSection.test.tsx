import { render, cleanup, screen } from "@testing-library/react";
import { GridSection } from "./GridSection";
import { IGridSectionProps } from "./IGridSectionProps";

describe("GridSection component", () => {
  let mock: IGridSectionProps;

  const testElements = ["Test1", "Test2"].map((item, i) => {
    return (
      <div data-testid="test-element" key={i}>
        {" "}
        {item}{" "}
      </div>
    );
  });

  beforeEach(() => {
    mock = {
      title: "Title",
      elements: testElements,
    };
  });

  afterEach(() => {
    cleanup();
  });

  test("GridSection should be render correctly", () => {
    const { container } = render(<GridSection {...mock} />);
    expect(container).toMatchSnapshot();
  });

  test("title should be present in document", () => {
    render(<GridSection {...mock} />);
    expect(screen.getByText(mock.title)).toBeInTheDocument();
  });

  test("number of rendered elements match with the length of the testElements array", () => {
    const { queryAllByTestId } = render(<GridSection {...mock} />);
    expect(queryAllByTestId("test-element")).toHaveLength(mock.elements.length);
  });
});
