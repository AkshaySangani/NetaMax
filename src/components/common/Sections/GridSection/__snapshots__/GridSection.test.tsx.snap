// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridSection component GridSection should be render correctly 1`] = `
<div>
  <div
    class="chakra-container css-tflc3m"
  >
    <div
      class="css-n19ytf"
    >
      <div
        class="css-tik1gu"
      >
        <p
          class="chakra-text css-1ebn8vi"
        >
          Title
        </p>
      </div>
      <div
        class="css-0"
      >
        <div
          class="css-1ymg3tg"
        >
          <div
            data-testid="test-element"
          >
             
            Test1
             
          </div>
          <div
            data-testid="test-element"
          >
             
            Test2
             
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
